builtin/providers/aws/resource_aws_iam_server_certificate.go

```diff
@@ -116,15 +120,15 @@ func resourceAwsIAMServerCertificateCreate(d *schema.ResourceData, meta interfac
 	}
 
 	log.Printf("[DEBUG] Creating IAM Server Certificate with opts: %s", createOpts)
-	resp, err := conn.UploadServerCertificate(createOpts)
+	_, err := conn.UploadServerCertificate(createOpts)
 	if err != nil {
 		if awsErr, ok := err.(awserr.Error); ok {
 			return fmt.Errorf("[WARN] Error uploading server certificate, error: %s: %s", awsErr.Code(), awsErr.Message())
 		}
 		return fmt.Errorf("[WARN] Error uploading server certificate, error: %s", err)
 	}
 
-	d.SetId(*resp.ServerCertificateMetadata.ServerCertificateId)
+	d.SetId(sslCertName)
```

Comment:
Hi @mrcopper 

Thanks for the work here - we can't just change the ID i'm afraid - we would need to migrate the users state otherwise everyone would show that changes are required

When we can do is to is to leave the Id as is and change the test to make sure it passes the correct value to the custom import func

The test would then look as follows:

```
func TestAccAWSIAMServerCertificate_importBasic(t *testing.T) {
 +	resourceName := "aws_iam_server_certificate.test_cert"
 +
 +	resource.Test(t, resource.TestCase{
 +		Providers:    testAccProviders,
 +		CheckDestroy: testAccCheckIAMServerCertificateDestroy,
 +		Steps: []resource.TestStep{
 +			resource.TestStep{
 +				Config: testAccIAMServerCertConfig,
 +			},
 +
 +			resource.TestStep{
 +				ResourceName:      resourceName,
 +				ImportState:       true,
 +				ImportStateVerify: true,
                                ImportStateId: "the id from the config for the name",
 +				ImportStateVerifyIgnore: []string{
 +					"private_key"},
 +			},
 +		},
 +	})
 +}
```
This would then allow the users to specify the name value for import

Hope this helps

Paul